orchestrator-salt/src/main/resources/salt/salt/postgresql/disaster_recovery/scripts/backup_db.sh

```diff
@@ -53,6 +53,13 @@ export PGSSLMODE=verify-full
 {%- endif %}
 
 errorExit() {
+  if [ -d "$DATE_DIR" ]; then
+    rm -f "$DATE_DIR/hive_backup"
+    rm -f "$DATE_DIR/ranger_backup"
+    rm -f "$DATE_DIR/profiler_agent_backup"
+    rm -f "$DATE_DIR/profiler_metric_backup"
+    rmdir -v "$DATE_DIR" > >(tee -a $LOGFILE) 2> >(tee -a $LOGFILE >&2)
```

Comment:
Thank you so much for checking on this and the suggestion @sxxgrc !! I actually started by using `rm -rf` in [the first](https://github.com/hortonworks/cloudbreak/pull/13780/commits/ed6a9de18aebf38ed84113463864da22f70dd101) commit but I changed it to this version because I think using `-rf` is so risky (though the chance is very rare)... for example, if something happened and the `$DATE_DIR ` becomes a very important directory and this command forces it to be removed, it would be a disaster. I don't think that case will happen but...just in case! Please let me know what you think!=)

Reply:
Here is the GAP i see in the implementation.
Case-1: The database backup is not limited to just hive, ranger and profiles databases. It can be used for other databases as well. We could invoke the script a specific database name as an input.
Case-2: We could extend the script to handle other databases as well. In that case we should make it a point to update this code as well.


in both these cases, we might still leak some data on failures.
@sxxgrc What do you think?

Reply:
Hmm, so let's maybe talk a bit about the idea of using `rm -rf` since that would account for both gaps Kalyan has mentioned without needing extra work.
Grace's concern is that the directory may be important, I want to provide some reasons I don't think this is the case:

- `DATE_DIR` is created by the backup script when it is run and is meant to be completely unique, using datetime, including seconds, to create the name of the directory. This means that the files in the directory can only be the ones created by the `backup_db.sh` script, no external forces, including the customer, should have any play here.
- In both the success case, and the failure case (which Grace has added) we are running `rmdir` with the expectation that the directory will be empty in order to delete any trace of the backup's local files. This makes me believe that the assumption is always that we want to remove the directory entirely.

So my argument is essentially that: The directory should only ever contain files created by the backup script, and leaving anything in it would be a data leak, so using `rm -rf` should pose no issues and in fact be more beneficial due to the cases defined by Kalyan.

Let me know what y'all think, I am open for anything but just want us to have a simple, efficient, and error-free solution.

Reply:
@sxxgrc @kkalvagadda1 Thank you both so much for the inputs! I think what Santiago explained above totally makes sense! I'm wondering if I can add maybe one more check to make sure this directory starts with `/var/tmp/` (so it's a temp folder for sure) and then force to remove it, in case in the future the code is changed and it accidentally removes important directories and files. We can use below code to achieve it. 
`[[ "$DATE_DIR" == "/var/tmp/"* ]] && rm -rf -v "$DATE_DIR" > >(tee -a $LOGFILE) 2> >(tee -a $LOGFILE >&2)`

Reply:
That's fine but if you're concerned and want to be sure you can just move the following 2 lines from the backup function to the top of the file so that they're for sure defined that way for the entire script:

```
BACKUPS_DIR="/var/tmp/"
DATE_DIR=${BACKUPS_DIR}/$(date '+%Y-%m-%dT%H:%M:%SZ')
```

Reply:
@sxxgrc This is a great idea!! I've moved them to the top to make them global variables. Please take a look at the updated code and let me know if you have any feedback. Thank you!!